### PR TITLE
add device backend check

### PR DESCRIPTION
**Pull request by AllentDan**

### CODE REVIEW

File: mmdeploy/apis/utils/utils.py [hunk 1]
```diff
@@ -2,7 +2,26 @@
...
+    """Check if device is appropriate for the backend.
+
+    Args:
+        deploy_cfg (str | mmcv.Config): Deployment config file.
+        device (str): A string specifying device type.
+    """
+    backend = get_backend_config(deploy_cfg)['type']
+    device_id = parse_device_id(device)
+    mismatch = dict(
+        tensorrt=lambda id: id == -1,
```
Comment (grimoire) [inline mmdeploy/apis/utils/utils.py:19]: what about other backends?

**Review by grimoire — APPROVED** [review]

File: mmdeploy/apis/utils/utils.py [hunk 2]
```diff
@@ -2,7 +2,25 @@
 import mmcv
 
 from mmdeploy.codebase import BaseTask, get_codebase_class, import_codebase
-from mmdeploy.utils import get_codebase, get_task_type
+from mmdeploy.utils import (get_backend_config, get_codebase, get_task_type,
+                            parse_device_id)
+
+
+def check_backend_device(deploy_cfg: mmcv.Config, device: str):
```
Comment (RunningLeon) [inline mmdeploy/apis/utils/utils.py:9]: may add ut for this func

File: mmdeploy/apis/utils/utils.py [hunk 3]
```diff
@@ -2,7 +2,25 @@
...
+    """Check if device is appropriate for the backend.
+
+    Args:
+        deploy_cfg (str | mmcv.Config): Deployment config file.
+        device (str): A string specifying device type.
+    """
+    backend = get_backend_config(deploy_cfg)['type']
```
Comment (RunningLeon) [inline mmdeploy/apis/utils/utils.py:16]: could use `get_backend`

**Review by RunningLeon — APPROVED** [review]
LGTM

**Pull request merged**